src/rhc.erl

```diff
@@ -203,13 +209,90 @@ fetch(Rhc, QueueName) ->
+                        binary()|
+                        {deleted, term(), binary(), non_neg_integer(), non_neg_integer()}|
+                        {binary(), non_neg_integer(), non_neg_integer()}}.
+fetch(Rhc, QueueName, internal) ->
+    fetch(Rhc, QueueName);
+fetch(Rhc, QueueName, internal_aaehash) ->
+    QParams = [{object_format, internal_aaehash}],
+    URL = fetch_url(Rhc, QueueName, QParams),
+    case request(get, URL, ["200"], [], [], Rhc) of
+        {ok, _Status, _Headers, Body} ->
+            case Body of
+                <<0:8/integer>> ->
+                    {ok, queue_empty};
+                <<1:8/integer, 1:8/integer,
+                    SegmentID:32/integer, SegmentHash:32/integer,
+                    TCL:32/integer, TombClockBin:TCL/binary,
+                    CRC:32/integer, ObjBin/binary>> ->
+                    {crc_check(CRC, ObjBin),
+                        {deleted,
+                            binary_to_term(TombClockBin),
+                            ObjBin,
+                            SegmentID, SegmentHash}};
+                <<1:8/integer, 0:8/integer,
+                    SegmentID:32/integer, SegmentHash:32/integer,
+                    CRC:32/integer, ObjBin/binary>> ->
+                    {crc_check(CRC, ObjBin),
+                        {ObjBin, SegmentID, SegmentHash}}
+            end;
+        {error, Error} ->
+            {error, Error}
+    end.
+
+-spec push(rhc(),
+            binary(),
+            [{riakc_obj:bucket(), riakc_obj:key(), riakc_obj:vclock()}]) ->
+                {error, term()}|{ok, iolist()}.
+push(Rhc, QueueName, KeyClockList) ->
+    URL = push_url(Rhc, QueueName),
+    ReqHeaders = [{"content-type", "application/json"}],
+    ReqBody = encode_keys_and_clocks(KeyClockList),
+    case request(post, URL, ["200"], ReqHeaders, ReqBody, Rhc) of
+        {ok, "200", _RspHeaders, RspBody} ->
+            {ok, RspBody};
+        {error, Error} ->
+            {error, Error}
+    end.
+
+
 fetch_url(Rhc, QueueName, Params) ->
     binary_to_list(iolist_to_binary([root_url(Rhc),
                                         "queuename/",
                                         mochiweb_util:quote_plus(QueueName),
                                         "?",
                                         mochiweb_util:urlencode(Params)])).
 
+push_url(Rhc, QueueName) ->
+    binary_to_list(
+        iolist_to_binary(
+            [root_url(Rhc),
+                "queuename/",
+                mochiweb_util:quote_plus(QueueName)])).
+
+-spec encode_keys_and_clocks([{riakc_obj:bucket(), riakc_obj:key(), riakc_obj:vclock()}]) -> iolist().
+encode_keys_and_clocks(KeysNClocks) ->
+    Keys = {struct, [{<<"keys-clocks">>,
+                      [{struct, encode_key_and_clock(Bucket, Key, Clock)} || {Bucket, Key, Clock} <- KeysNClocks]
+                     }]},
+    mochijson2:encode(Keys).
+
+encode_key_and_clock({Type, Bucket}, Key, Clock) ->
+    [{<<"bucket-type">>, Type},
```

Comment:
Here is how we name the key for `bucket-type`

Reply:
Acknowledged - see https://github.com/basho/riak_kv/pull/1792